src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -2756,12 +2635,11 @@ private void handleStateNormal(final InetAddressAndPort endpoint, final String s
             logger.warn("Node {} is currently being replaced by node {}.", endpoint, replacementNode.get());
         }
 
-        updatePeerInfo(endpoint);
```

Comment:
How is this update happening now? Seems to be missing a call to pass the new data into Nodes?

Reply:
Whenever a new endpoint state is added to the map, it gets an updater and propagates all the changes immediately: see those methods: https://github.com/datastax/cassandra/pull/300/files#diff-99267a2170b04fd7dd24d6c6bf2ba1fc26d6dc896cd74f8c5bd56c476e2540e4R2288 and then https://github.com/datastax/cassandra/pull/300/files#diff-398cb35f0a0e84f5142f64c40ca2d0280803e4b2b39478409d2656ff8e594a69R167